doc/controller_configuration/controller_configuration_tutorial.rst

```diff
@@ -30,146 +79,409 @@ The ``controllers.yaml`` configuration file is located in the ``robot_moveit_con
+  * declares ``moveit_ros_control_interface::ControllerHandleAllocator`` class
+
+* ``#include <pluginlib/class_list_macros.h>``
+
+  * declares ``PLUGINLIB_EXPORT_CLASS`` macro for exporting plugins
+
+Two example *controller handle* implementations are included with MoveIt:
+
+* `follow_joint_trajectory_controller_handle.h <https://github.com/ros-planning/moveit/blob/master/moveit_plugins/moveit_simple_controller_manager/include/moveit_simple_controller_manager/follow_joint_trajectory_controller_handle.h>`_
+
+  * See implementation in `follow_joint_trajectory_controller_handle.cpp <https://github.com/ros-planning/moveit/blob/master/moveit_plugins/moveit_simple_controller_manager/src/follow_joint_trajectory_controller_handle.cpp>`_
+
+* `gripper_controller_handle.h <https://github.com/ros-planning/moveit/blob/master/moveit_plugins/moveit_simple_controller_manager/include/moveit_simple_controller_manager/gripper_controller_handle.h>`_
+
+  * Implemented inline in the same header file
+
+As you can see, writing a `controller handle <https://github.com/ros-planning/moveit/blob/master/moveit_core/controller_manager/include/moveit/controller_manager/controller_manager.h#L104>`__ comes down to implementing:
+
+* ``sendTrajectory`` method that translates `moveit_msgs::RobotTrajectory <http://docs.ros.org/en/noetic/api/moveit_msgs/html/msg/RobotTrajectory.html>`_ to a format the controller can understand
+* ``cancelExecution`` method to tell the controller to stop any active trajectories
+* ``waitForExecution`` method that will block the calling thread until the controller finishes or the ``timeout`` is reached
+* ``getLastExecutionStatus`` method that returns the status of the last requested trajectory.
+
+One example *controller handle allocator* plugin implementation is included with MoveIt:
+
+* `joint_trajectory_controller_plugin.cpp <https://github.com/ros-planning/moveit/blob/master/moveit_plugins/moveit_ros_control_interface/src/joint_trajectory_controller_plugin.cpp>`_
+
+The only job of a controller handle allocator is to create a new instance of the controller handle. The following example implements an allocator for a custom controller handle of type ``example::controller_handle_example``:
+
+.. code-block:: c++
+
+  // declares example::controller_handle_example class
+  #include "controller_handle_example.h"
+  #include <moveit_ros_control_interface/ControllerHandle.h>
+  #include <pluginlib/class_list_macros.h>
+
+  namespace example
+  {
+  class controller_handle_allocator_example : public moveit_ros_control_interface::ControllerHandleAllocator
+  {
+  public:
+    moveit_controller_manager::MoveItControllerHandlePtr alloc(const std::string& name,
+                                                               const std::vector<std::string>& resources) override
+    {
+      return std::make_shared<controller_handle_example>(name, std::string("follow_joint_trajectory"));
```

Comment:
The constructor API shoud not include the `follow_joint_trajectory` string. This controller type doesn't need extra work ;-)

Reply:
Simplified code samples so this parameter is not used. Re-built code to verify it compiles correctly.